security/configure-with-vault/README.md

```diff
@@ -0,0 +1,409 @@
+authorized to read credentials from Vault.
+
+```
+podTemplate:
+    ...
+    annotations:
+      vault.hashicorp.com/agent-inject: "true"
+      vault.hashicorp.com/agent-inject-status: update
+      vault.hashicorp.com/preserve-secret-case: "true"
+      vault.hashicorp.com/agent-inject-secret-jksPassword.txt: secret/jksPassword.txt
+      vault.hashicorp.com/agent-inject-template-jksPassword.txt: |
+        {{- with secret "secret/jksPassword.txt" -}}
+        {{ .Data.data.password }}
+        {{- end }}
+      ...
+      vault.hashicorp.com/role: confluent-operator
+```
+
+The above annotations will trigger all the magic that will result in the Vault secrets 
+being written to an in-memory filesystem dynamically mounted to the Confluent Platform 
+component worker container. The format of the files will match what’s required by 
+Confluent Platform.
+
+
+## Deploy Confluent Platform (with RBAC)
+
+Read the main scenario example documentation to understand the concepts:
+https://github.com/confluentinc/confluent-kubernetes-examples/tree/master/security/production-secure-deploy
+
+Note: There are two differences from the above scenario, when using "Directory in path container":
+
+- You'll need to use Kubernetes secrets for the KafkaRestClass authentication. Confluent for Kubernetes 
+2.0.x does not support using directory path in container for this specific credential.
+- You'll need to specify the RBAC rolebindings for the CP components. These will not be created 
+automatically.
+
+Create a KafkaRestClass object with a user that has cluster access to create rolebindings 
+for Confluent Platform RBAC. In this scenario, that is user `kafka`:
+
+```
+kubectl -n confluent create secret generic rest-credential \
+  --from-file=bearer.txt=$TUTORIAL_HOME/credentials/rbac/kafkarestclass/bearer.txt
+
+kubectl -n confluent apply -f $TUTORIAL_HOME/rbac/kafka-rest.yaml
+```
+
+### Deploy OpenLDAP
+
+This repo includes a Helm chart for [OpenLdap](https://github.com/osixia/docker-openldap). 
+The chart `values.yaml` includes the set of principal definitions that Confluent Platform 
+needs for RBAC.
+
+Deploy OpenLdap:
+
+```
+helm upgrade --install -f $TUTORIAL_HOME/../../assets/openldap/ldaps-rbac.yaml test-ldap $TUTORIAL_HOME/../../assets/openldap --namespace confluent
+```
+
+Validate that OpenLDAP is running:  
+
+```
+kubectl get pods -n confluent
+```
+
+### Deploy Confluent Platform
+
+Deploy Zookeeper and Kafka first:
+
+```
+kubectl apply -f $TUTORIAL_HOME/rbac/zk_kafka.yaml --namespace confluent
+```
+
+Create the RBAC Rolebindings needed for the CP components:
+
+```
+kubectl apply -f $TUTORIAL_HOME/rbac/internal-rolebinding.yaml --namespace confluent
+```
+
+Deploy the CP components:
+
+```
+kubectl apply -f $TUTORIAL_HOME/rbac/cp_component.yaml --namespace confluent
+```
+
+
+## Tear Down
+
+```
+kubectl delete -f $TUTORIAL_HOME/confluent-platform-norbac-vault.yaml -n confluent
```

Comment:
missing teardown for the internal-rolebinding.yaml and kafka-rest.yaml.

Reply:
Added!